rotkehlchen/chain/aggregator.py

```diff
@@ -1286,72 +1287,157 @@ def get_evm_manager(
+                        continue
+                    chains_to_check.append(chain)
+
+                added_accounts += self.check_chains_and_add_accounts(
+                    account=account,
+                    chains=chains_to_check,
                 )
-            except InputError:
-                log.debug(f'Not adding {account} to {chain} since it already exists')
+
+        return added_accounts
+
+    def detect_evm_accounts(
+            self,
+            progress_handler: Optional['ProgressUpdater'] = None,
+    ) -> list[tuple[SUPPORTED_EVM_CHAINS, ChecksumEvmAddress]]:
+        """
+        Detects user's EVM accounts on different chains and adds them to the tracked accounts.
+        1. Iterates through already added addresses
+        2. Checks whether the address is the chain that it is currently added to
+        3. If it is not, checks whether the address is active in a chain
+        4. If it is active, adds it to the tracked accounts on that chain
+
+        Returns a list of tuples of (chain, address) for the freshly detected accounts.
+        """
+        current_accounts: dict[ChecksumEvmAddress, list[SUPPORTED_EVM_CHAINS]] = defaultdict(list)
+        chain: SUPPORTED_EVM_CHAINS
+        for chain in typing.get_args(SUPPORTED_EVM_CHAINS):
+            chain_accounts = self.accounts.get(chain)
+            for account in chain_accounts:
+                current_accounts[account].append(chain)
+
+        all_evm_chains = set(typing.get_args(SUPPORTED_EVM_CHAINS))
+        added_accounts: list[tuple[SUPPORTED_EVM_CHAINS, ChecksumEvmAddress]] = []
+        for account, chains in current_accounts.items():
+            if progress_handler is not None:
+                progress_handler.new_step(f'Checking {account} EVM chain activity')
+
+            if self.is_contract(account, chains[0]):
+                continue  # do not check contracts
+
+            chains_to_check = list(all_evm_chains - set(chains))
+            if len(chains_to_check) == 0:
                 continue
-            added_accounts.append((chain, account))
+
+            added_accounts += self.check_chains_and_add_accounts(account, chains_to_check)
+
+        if progress_handler is not None:
+            progress_handler.new_step('Potentially write migrated addresses to the DB')
+
+        with self.database.user_write() as write_cursor:
+            self.database.add_blockchain_accounts(
+                write_cursor=write_cursor,
+                account_data=[
+                    BlockchainAccountData(chain, account)
+                    for chain, account in added_accounts
+                ],  # not duplicating label and tags as it's chain specific
+            )
+
+        with self.database.user_write() as cursor:
+            cursor.execute(
+                'INSERT OR REPLACE INTO multisettings (name, value) VALUES (?, ?)',
+                (EVM_ACCOUNTS_DETECT_KEY, str(ts_now())),
+            )
```

Comment:
I made the same mistake, if you look at this table name and value are not primary keys so this will keep inserting rows with the same name instead of keeping only one. We said to use the settings table for now and then we can create a new table for this and clean the settings table

Reply:
Thanks for pointing it out!